lib/credentials_from_file.c

```diff
@@ -91,122 +119,46 @@ bool write_normal_device_credentials(const char *dev_cred_file,
  */
 
 bool write_secure_device_credentials(const char *dev_cred_file,
-				     sdo_sdk_blob_flags flags,
-				     sdo_dev_cred_t *ocred)
+				     sdo_sdk_blob_flags flags, sdo_dev_cred_t *ocred)
 {
 	bool ret = true;
-
-	(void)ocred; /* Unused warning */
+	(void) *ocred;
 
 	if (!dev_cred_file) {
 		return false;
 	}
 
 #ifndef NO_PERSISTENT_STORAGE
-	sdow_t sdowriter, *sdow = &sdowriter;
 
-	if (!sdow_init(sdow)) {
+	sdow_t *sdow = sdo_alloc(sizeof(sdow_t));
+	if (!sdow_init(sdow) || !sdo_block_alloc(&sdow->b) || !sdow_encoder_init(sdow)) {
 		LOG(LOG_ERROR, "sdow_init() failed!\n");
-		return false;
+		ret = false;
+		goto end;
 	}
-/*
-	sdow_begin_object(sdow);
-	sdo_write_tag(sdow, "Secret");
-	sdow_begin_sequence(sdow);
-*/
 	sdo_byte_array_t **ovkey = getOVKey();
-
 	if (!ovkey || !*ovkey) {
 		ret = false;
 		goto end;
 	}
-	/*
-	sdo_write_byte_array_field(sdow, (*ovkey)->bytes, INITIAL_SECRET_BYTES);
-	sdow_end_sequence(sdow);
-	sdow_end_object(sdow);
-	*/
-
-	sdow_start_array(sdow, 1);
 	sdow_byte_string(sdow, (*ovkey)->bytes, INITIAL_SECRET_BYTES);
-	sdow_end_array(sdow);
-
-	/* Fill sdow buffer */
-
-	if (sdo_blob_write((char *)dev_cred_file, flags, &sdow->b.block[0],
-			   sdow->b.block_size) == -1) {
-		LOG(LOG_ERROR, "Issue while writing Devcred blob\n");
+	size_t encoded_secret_length = 0;
+	if (!sdow_encoded_length(sdow, &encoded_secret_length) || encoded_secret_length <= 0) {
```

Comment:
Same as above (regarding negative value for size_t).

Reply:
Done.